src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,278 @@
+      return this.backspace ? this.numeralCaretOnBackspace(selection, text)
+        : this.delete ? this.numeralCaretOnDelete(selection, text)
+          : this.numeralCaretOnInsert(selection)
+    },
+    numeralCaretOnInsert (selection) {
+      selection -= this.selection < selection ? 1 : 0
+      selection -= this.selection === selection ? 1 : 0
+
+      return selection
+    },
+    numeralCaretOnDelete (selection, text) {
+      if (text.substr(this.selection, 1) === this.decimal) {
+        selection += 1
+      } else {
+        selection += text.substr(this.numeralPrefix.length + this.sign.length, 1) === '0' ? 1 : 0
+        selection += text.substr(selection - 1, 1) === this.decimal ? 1 : 0
+      }
+
+      return selection
+    },
+    numeralCaretOnBackspace (selection, text) {
+      if (text.substr(this.numeralPrefix.length + this.sign.length, 1) === '0') {
+        // abs(value) < 1
+        text.substr(this.selection, 1) !== this.decimal && selection++
+        text.substr(this.selection - 1, 1) === this.decimal && selection++
+      }
+
+      return selection
+    },
+    // Correction on arbitrary value entered externally
+    attemptNumeralCorrection (text) {
+      const decimalPoint = text.lastIndexOf('.')
+      let integer
+
+      // Integer only
+      if (!this.precision) {
+        // Assume floor conversion (truncate)
+        return decimalPoint < 0
+          ? this.removeNonNumeral(text) : this.removeNonNumeral(text.substring(0, decimalPoint))
+      }
+
+      // Missing fractional part
+      if (decimalPoint < 0) {
+        integer = this.removeNonNumeral(text)
+        if (!integer) return null // garbage
+
+        return integer + this.getFraction('')
+      }
+
+      integer = this.removeNonNumeral(text.substring(0, decimalPoint))
+      if (!integer) integer = '0'
+
+      return integer + this.getFraction(text.substr(decimalPoint))
+    },
+    getFraction (text) {
+      let fraction = this.removeNonNumeral(text)
+      fraction = fraction.substr(0, this.precision)
+      fraction = fraction.padEnd(this.precision, '0')
+
+      return fraction
+    },
+    addPrefixSuffix (text) {
+      return this.numeralPrefix + this.sign + text + this.numeralSuffix
+    },
+    parseInt (text) {
+      return this.removeNonNumeral(this.removeLeadingZeros(text))
+    },
+    parseFloat (text) {
+      text = this.removeLeadingZeros(text)
+
+      const decimal = text.lastIndexOf('.')
+      const fraction = this.getFraction(text.substr(decimal + 1))
+
+      return text.substr(0, decimal) + '.' + fraction
+    },
+    removeLeadingZeros (text) {
+      let parsed = []
+      let leadingZeros = true
+
+      for (const digit of text) {
+        if (leadingZeros) {
+          if (digit === '0') continue
+          leadingZeros = false
+        }
+
+        parsed.push(digit)
+      }
+
+      parsed = parsed.join('')
+      return parsed[0] === '.' ? '0' + parsed : parsed
```

Comment:
Does it assume that decimal point is always a dot?

Reply:
It's always a dot in `this.lazyValue`.